lib/Dialect/Arc/ArcOps.cpp

```diff
@@ -142,25 +142,28 @@ void DefineOp::print(OpAsmPrinter &p) {
 }
 
 LogicalResult DefineOp::verifyRegions() {
+
   // Check that the body does not contain any side-effecting operations. We can
   // simply iterate over the ops directly within the body; operations with
   // regions, like scf::IfOp, implement the `HasRecursiveMemoryEffects` trait
   // which causes the `isMemoryEffectFree` check to already recur into their
   // regions.
-  for (auto &op : getBodyBlock()) {
-    if (isMemoryEffectFree(&op))
-      continue;
-
-    // We don't use a op-error here because that leads to the whole arc being
-    // printed. This can be switched of when creating the context, but one
-    // might not want to switch that off for other error messages. Here it's
-    // definitely not desirable as arcs can be very big and would fill up the
-    // error log, making it hard to read. Currently, only the signature (first
-    // line) of the arc is printed.
-    auto diag = mlir::emitError(getLoc(), "body contains non-pure operation");
-    diag.attachNote(op.getLoc()).append("first non-pure operation here: ");
-    return diag;
-  }
+  // Q. Can we allow side-effecting ops within Arc?
+  // for (auto &op : getBodyBlock()) {
+  //   if (isMemoryEffectFree(&op))
+  //     continue;
+
+  //   // We don't use a op-error here because that leads to the whole arc being
+  //   // printed. This can be switched of when creating the context, but one
+  //   // might not want to switch that off for other error messages. Here it's
+  //   // definitely not desirable as arcs can be very big and would fill up the
+  //   // error log, making it hard to read. Currently, only the signature
+  //   (first
+  //   // line) of the arc is printed.
+  //   auto diag = mlir::emitError(getLoc(), "body contains non-pure
+  //   operation"); diag.attachNote(op.getLoc()).append("first non-pure
+  //   operation here: "); return diag;
+  // }
```

Comment:
Clearly it's not correct. How can we model a state transfer function that has side-effect? Do we need a new op?